File: hexameter-core/src/test/java/org/codetome/hexameter/core/api/DefaultSatelliteDataTest.java
package org.codetome.hexameter.core.api;

import org.codetome.hexameter.core.api.defaults.DefaultSatelliteData;
import org.junit.Assert;
import org.junit.Before;
import org.junit.Test;

public class DefaultSatelliteDataTest {

    private static final double EXPECTED_MOVEMENT_COST = 5.1;
    private static final boolean EXPECTED_IS_PASSABLE = true;
    private static final boolean EXPECTED_IS_BLOCKS_VIEW = true;

    private DefaultSatelliteData target;


    @Before
    public void setUp() throws Exception {
        target = new DefaultSatelliteData() {
        };
    }

    @Test
    public void shouldProperlySetAndGetIsPassable() {
        target.setPassable(EXPECTED_IS_PASSABLE);
        Assert.assertEquals(target.isPassable(), EXPECTED_IS_PASSABLE);
    }

    @Test
    public void shouldProperlySetAndGetIsBlocksView() {
        target.setOpaque(EXPECTED_IS_BLOCKS_VIEW);
        Assert.assertEquals(target.isOpaque(), EXPECTED_IS_BLOCKS_VIEW);
    }

    @Test
    public void shouldProperlySetAndGetMovementCost() {
        target.setMovementCost(EXPECTED_MOVEMENT_COST);
        Assert.assertEquals(EXPECTED_MOVEMENT_COST, target.getMovementCost(), 0);
    }
}

File: mixite2.core/src/commonTest/kotlin/org/hexworks/mixite2/core/api/DefaultSatelliteDataTest.kt
package org.hexworks.mixite2.core.api

import org.hexworks.mixite2.core.api.defaults.DefaultSatelliteData
import kotlin.test.BeforeTest
import kotlin.test.Test
import kotlin.test.assertEquals

class DefaultSatelliteDataTest {

    lateinit var target: DefaultSatelliteData


    @BeforeTest
    fun setUp() {
        target = object : DefaultSatelliteData() {

        }
    }

    @Test
    fun shouldProperlySetAndGetIsPassable() {
        target.isPassable = EXPECTED_IS_PASSABLE
        assertEquals(EXPECTED_IS_PASSABLE, target.isPassable)
    }

    @Test
    fun shouldProperlySetAndGetIsBlocksView() {
        target.isOpaque = EXPECTED_IS_BLOCKS_VIEW
        assertEquals(EXPECTED_IS_BLOCKS_VIEW, target.isOpaque)
    }

    @Test
    fun shouldProperlySetAndGetMovementCost() {
        target.movementCost = EXPECTED_MOVEMENT_COST
        assertEquals(EXPECTED_MOVEMENT_COST, target.movementCost)
    }

    companion object {

        private const val EXPECTED_MOVEMENT_COST = 5.1
        private const val EXPECTED_IS_PASSABLE = true
        private const val EXPECTED_IS_BLOCKS_VIEW = true
    }
}
